### ISSUE TITLE
Users spontaneously being logged out

### ISSUE THREAD

**Issue by lostamongtrees** [RolePlayGateway/roleplaygateway#192]
Bug initially reported in the chat by Wolf's Bane109

> everytime I try to submit a post I keep getting logged out. Anyone know why that is?
> It's a glitch with tagging characters. i was able to submit it fine without tagging characters. that's just annoying for people that make long posts and don't copy their posts. Thankfully I did.

going to attempt to replicate

**Current Possibilities:**
- [ ] Tagging characters logs you out on post submit
- [ ] Trying to submit characters
- [ ] Editing characters
- [ ] Attempting to post
- [ ] Just browsing the site
- [ ] Basically doing anything ever

**Other information:**
- Seemingly logged out less when using Safari on iPad or iPhone?
- Worsens with multiple tabs open?

**Comment by lostamongtrees** [comment]
i was unable to replicate the issue

**Comment by lostamongtrees** [comment]
Might have something to do with #195

**Comment by Lady-Awinita** [comment]
Random logouts still occur for me when doing the following

- Visiting notifications
- Visiting snippets linked in new chat
- Visiting profiles on site directly.

End result however redirects on login to the main forums page instead of the page visited forcing me to go back to said page instead of logging back in and refreshing back to where I was

EDIT: Tagging characters in posts now works, albiet the site is slow on the loading the page {it does get to the "You have Replied to this roleplay" page just fine but once past that it takes a few secnds longer than it should to load the newest post in the place {logout does not occur}}

**Comment by shentino** [comment]
Same happening here

**Comment by martindale** [comment]
> Same happening here

What is "same"?  What specific action caused your most recent logout, and can you replicate the issue?  Please see instructions for helping [in this thread on the forum](https://www.roleplaygateway.com/getting-logged-out-t112192.html)...

**Comment by martindale** [comment]
@shentino do you mind checking in on the case reported in #218?

**Comment by Lady-Awinita** [comment]
For me my exact actions was the following path.

- Character Management --> Drop down menu link COMMUNITY
- resulted in getting logged out enroute to the main forums page for Community.

This has begun to become a nightly thing every 24 hours or so for me around the time between 7-10PM eastern american time

**Comment by martindale** [comment]
Awi, does the logout happen every time you do this most recent step, or
only during those specific times?  If you can duplicate it reliably outside
of those hours, we can save a lot of time.

Alternatively, help us out with a screenshot of the "Cookies" section in
your browser inspector's "Resources" tab via the beta channel on RPG Chat.
Thanks!

P.S., it's usually CTRL+SHIFT+I to launch the inspector, but only do so
whilst first logged in on RPG's front page.  You can DuckDuckGo "viewing
site cookies" + your browser name for more detailed instructions.

**Comment by Lady-Awinita** [comment]
Tried that and got lost because its not the same as it used to be. I cant make head or tail of what I am supposed to look for however Chrome is throwing a warning that a cookie was blocked. Apparently the cookies now require SameSite to be enabled

**Comment by Lady-Awinita** [comment]
> Awi, does the logout happen every time you do this most recent step, or
> only during those specific times?  If you can duplicate it reliably outside
> of those hours, we can save a lot of time.

It seems to happen mostly between those times stated, but next time it happens I will post here letting you know

**Comment by Lady-Awinita** [comment]
Current report; it has been 48 hours since my last logout. this was the steps taken for the logout to occur
- Forums page then clicking to
- Private Messages
- TIME OF LOGOUT: 7:58PM eastern time

**Comment by Lady-Awinita** [comment]
Just got logged out when reloading a page, and instead of reloading the place page I was at, it redirects to the main forums when it SHOULD just reload the place page I was at

**Comment by Lady-Awinita** [comment]
And now I am logged ourt when trying to post.

**Comment by Lady-Awinita** [comment]
Because I was asked list my currently active extensions I shall do so here

- Dark Reader; this basically turns the website to a dark easier to read theme
- Stylish; same thing but more focused on google to a darker theme or custom backgrounds
- Tampermonkey; a codec designed for a chatroom through deviantart
- uBlock Origin; my adblock software

**Comment by lostamongtrees** [comment]
I've noticed that in the explorer, `401` 's pop up very consistently. 
i suspect the log out bug has something to do with the cache
`cache-control	
no-cache, no-store, must-revalidate`

`401` Happening every 7-9 rows in the inspector (sprinkled between `304` and `200` with every now and again a `502` showing up

**Comment by lostamongtrees** [comment]
I think it has something to do with bandwidth?

I noticed a few things
- Pages with lots of data tend to trigger log out
-- awi loading characters
-- nore reloading the multiverse page
-- explorer loading dense locations
--- not sure if character density or post density
-- if you're logged in across too many tabs
-- someone noted it happens after a certain period of time, which seems to shorten the more someone uses it

also

I was trying to load
https://www.roleplaygateway.com/universes/the-multiverse/map

at the same time as
https://www.roleplaygateway.com/
Which loaded fine. I logged in, and went straight to `characters` , clicked `play` and was instantly whitescreened.
https://www.roleplaygateway.com/explorer/?characterID=262372&roleplayID=1

Checking with another tab, I had been logged out.
Coincidentally, `/maps` decided to load in the moments between me loading the characters page and me clicking play.

cache acting funky?

**Comment by shentino** [comment]
Why is the cache even affecting things in the first place?

Shouldn't session information be stored in a cookie or in an sid parameter?

**Comment by shentino** [comment]
As a design principle, anything variant shouldn't be cacheable.  The only things that should be cached are static assets like avatars, media, maps, and the like.  Text content, and especially login credentials, should not be cached.

**Comment by lostamongtrees** [comment]
Had an experiment, on if logging in was impacted by increased activity in another tab

- 2 tabs open
- In one tab I was attempting a log in from the designated logging in page.
- In another, second tab, I was whitescreened and refreshing at constant. refreshrefreshrefresh
- When the first tab loaded, slower than normal, it was back on the front page and **not** logged in.

and then

- I went to log in again, this time not touching anything while it logged in, and it redirected to the front page logged in.
- I then refreshed the other tab and the explorer loaded as it should.

**Comment by Lady-Awinita** [comment]
CURRENTLY EXP{ERIMENTING but so far things are working, I havent been logged out in 6 days now since disabling Dark Reader

HOWEVER

I am not the only one who was having this logout issue constantly, Ritnio cant do anything on the site, page to page, vidst his control panel, anything, without getting logged out. He is constantly logging back in. he reported no extensions active

**Comment by lostamongtrees** [comment]
Barney_Fife noticed that it is nearly impossible to stay logged in via mobile
- still leaning on data/bandwidth

@Lady-Awinita  is Ritnio a mobile user?

**Comment by lostamongtrees** [comment]
i notice i dont get logged out when using my mobile hotspot nearly as much as when connected to different wifis. 

- TMOBILE I barely get hit with the bug but when I do it's always because I'm trying to load too much at once, that is a constant. I do notice getting hit with it more or less depending on how I am accessing the internet. I haven't myself gotten logged out on mobile & seem to have little to no issue while using my hotspot
- COMCAST  The overload log outs tend to most happen when I am at my friend's house. I believe he has comcast but can doublecheck.
- AT&T At home I barely notice the issue / don't seem impacted by it but i'm not home too much. AT&T.

**Comment by LawOfTheLand** [comment]
Workaround as reported by Ritnio:

If you're getting logged out while trying to post forum style, do this:

Post
If asked to log in, do so
Hit back twice
Select character tags, then post

This not only lets the post go through, but avoids mandating Tag All.

**Comment by Lady-Awinita** [comment]
> Workaround as reported by Ritnio:
> 
> If you're getting logged out while trying to post forum style, do this:
> 
> Post
> If asked to log in, do so
> Hit back twice
> Select character tags, then post
> 
> This not only lets the post go through, but avoids mandating Tag All.

What about characters not PHYSICALLY in that location ? typing in their name when making the post will the post go through without getting logged out ?

EDIT: This is one of the reasons why I ghyavent posted in so long also burnout is affecting me at this moment too

**Comment by lostamongtrees** [comment]
Was logged out on tab while submitting a private message in another tab. 
Replicated the issue

**Comment by Kura-Ravengade** [comment]
Does anyone know when this will be fixed??

**Comment by martindale** [comment]
Rough times tracking this down, having spent _many_ hours trying to isolate replicable cases — it seems almost entirely random, and since the code to manage user sessions hasn't been touched in more than a decade, I'm leaning towards this being a more complicated database issue (sessions corrupted due to FPM aborting requests?  Session table exceeding quota?) rather than a specific bug in the code.

We're still working on it, but without any way to replicate it on demand, it's like finding a needle in a haystack.

**Comment by partially-stars** [comment]
Just wanted to update- a lot of users are still reporting issues with being logged out. Tagging characters still seems to be an issue, after writing posts and tagging characters I have to make sure I’m logged in from a different tab because the vast, vast majority of times I will have been logged out. Other users have reported having to log in multiple times for the log in to stick, and being logged out just when moving between sites. The issues raised above still seem to apply, i.e. they’re not as bad on mobile and iPad, and having multiple tabs open seems to worsen the issue.

**Comment by martindale** [comment]
Thanks for reporting this, @partially-stars!  We're focusing in on character tagging and posting from the main site (as opposed to the Explorer).  No clear leads as of yet, but I'll report back as soon as I have something!

**Comment by martindale** [comment]
As reported on tonight's call, I've made another internal change w.r.t. "debug" being enabled on various pages; **this may or may not resolve the issue**, but I've got my fingers crossed that my hunch was correct.  Please report back if the issue has resolved itself (i.e., actions which previously reliably logged you out no longer do so) or if it happens to you again!

**Comment by martindale** [comment]
Thanks to @partially-stars in [RPG Chat](https://chat.roleplaygateway.com/#/room/#help:roleplaygateway.com), it looks like we've fixed the issue for at least one person:
> Yep, managed to tag and post in one of my RPs without being logged out!

If we can get a few confirmations from the others who were experiencing this issue that it is indeed fixed, I'll close this issue and throw a little party. 🎉 

For those of you who were able to reliably reproduce the logout issue, if you could please test it now and let us know that'd be 🔥 — thanks to everyone who helped us get to this point! 🙇🙇🙇

**Comment by martindale** [comment]
We've got confirmations from multiple people that actions which would previously log them out are no longer doing so.  Thanks to everyone who helped troubleshoot this!  Closing and organizing a party.

**Comment by lostamongtrees** [comment]
### **YAAAAAAAAAAAS**
